pkg/drivers/vmware/driver.go

```diff
@@ -0,0 +1,775 @@
+			return err
+		}
+
+		// read generated public ssh key
+		if keyfh, err = os.Open(d.publicSSHKeyPath()); err != nil {
+			return err
+		}
+		defer keyfh.Close()
+
+		if keycontent, err = ioutil.ReadAll(keyfh); err != nil {
+			return err
+		}
+
+		// add public ssh key to authorized_keys
+		if err := executeSSHCommand(fmt.Sprintf("echo '%s' > /home/%s/.ssh/authorized_keys", string(keycontent), d.SSHUser), d); err != nil {
+			return err
+		}
+
+		// make it secure
+		if err := executeSSHCommand(fmt.Sprintf("chmod 600 /home/%s/.ssh/authorized_keys", d.SSHUser), d); err != nil {
+			return err
+		}
+
+		log.Debugf("Leaving create sequence early, configdrive found")
+		return nil
+	}
+
+	// Generate a tar keys bundle
+	if err := d.generateKeyBundle(); err != nil {
+		return err
+	}
+
+	// Test if /var/lib/boot2docker exists
+	vmrun("-gu", B2DUser, "-gp", B2DPass, "directoryExistsInGuest", d.vmxPath(), "/var/lib/boot2docker")
+
+	// Copy SSH keys bundle
+	vmrun("-gu", B2DUser, "-gp", B2DPass, "CopyFileFromHostToGuest", d.vmxPath(), d.ResolveStorePath("userdata.tar"), "/home/docker/userdata.tar")
+
+	// Expand tar file.
+	vmrun("-gu", B2DUser, "-gp", B2DPass, "runScriptInGuest", d.vmxPath(), "/bin/sh", "sudo sh -c \"tar xvf /home/docker/userdata.tar -C /home/docker > /var/log/userdata.log 2>&1 && chown -R docker:staff /home/docker\"")
+
+	// copy to /var/lib/boot2docker
+	vmrun("-gu", B2DUser, "-gp", B2DPass, "runScriptInGuest", d.vmxPath(), "/bin/sh", "sudo /bin/mv /home/docker/userdata.tar /var/lib/boot2docker/userdata.tar")
+
+	// Enable Shared Folders
+	vmrun("-gu", B2DUser, "-gp", B2DPass, "enableSharedFolders", d.vmxPath())
+
+	shareName := "Users"
+	shareDir := "/Users"
+
+	if shareDir != "" && !d.NoShare {
+		if _, err := os.Stat(shareDir); err != nil && !os.IsNotExist(err) {
+			return err
+		} else if !os.IsNotExist(err) {
+			// add shared folder, create mountpoint and mount it.
+			vmrun("-gu", B2DUser, "-gp", B2DPass, "addSharedFolder", d.vmxPath(), shareName, shareDir)
+			command := "[ ! -d " + shareDir + " ]&& sudo mkdir " + shareDir + "; sudo mount --bind /mnt/hgfs/" + shareDir + " " + shareDir + " || [ -f /usr/local/bin/vmhgfs-fuse ]&& sudo /usr/local/bin/vmhgfs-fuse -o allow_other .host:/" + shareName + " " + shareDir + " || sudo mount -t vmhgfs -o uid=$(id -u),gid=$(id -g) .host:/" + shareName + " " + shareDir
+			vmrun("-gu", B2DUser, "-gp", B2DPass, "runScriptInGuest", d.vmxPath(), "/bin/sh", command)
+		}
+	}
+	return nil
+}
+
+func (d *Driver) Start() error {
+	vmrun("start", d.vmxPath(), "nogui")
+
+	// Do not execute the rest of boot2docker specific configuration, exit here
+	if d.ConfigDriveURL != "" {
+		log.Debugf("Leaving start sequence early, configdrive found")
+		return nil
+	}
+
+	log.Debugf("Mounting Shared Folders...")
+	var shareName, shareDir string // TODO configurable at some point
+	switch runtime.GOOS {
+	case "darwin":
+		shareName = "Users"
+		shareDir = "/Users"
+		// TODO "linux" and "windows"
```

Comment:
Does not seem to provide a cross-platform solution. This will be one of the first questions we will get. How to get the shared folders working.

Reply:
We can follow the same pattern as virtualbox, use `C:\Users` for windows, `/home` for Linux.

Reply:
It is fixed.